### ISSUE TITLE
Python 3 upgrade

### ISSUE THREAD

**Issue by NickSwainston** [JLBLine/srclists#4]
A python 3 version of srclist_by_beam.py would be convenient. We just need to change to python 2 modules to run it currently which isn't too hard so no rush.

**Comment by JLBLine** [comment]
Hey Nick, I just looked into this, and the problem is that this module relies on either mwapy or mwa_py - some uses still rely on the mwapy from MWA_TOOLS. And until the master branch of mwa_pb is merged with the 2to3 branch on mwa_py, none of the beam code can run with python3. So until that get's fixed up, we're stuck

**Comment by NickSwainston** [comment]
No worries, I do have a python 3 branch of mwa_py but only the analytical beam works. The full EE beam breaks in python 3 so, as you say, we'll have to wait on someone who understands the code to fix it.

**Comment by cjordan** [comment]
@NickSwainston - please give the current of srclists master a try. In conjunction with a change I made to the 2to3 branch of mwa_pb, I think things are working.

**Comment by NickSwainston** [comment]
Assuming I've correctly installed the latest version of mwa_pb, I'm still getting errors:

```
> python3 srclist_by_beam.py -m tests/test_files/1173782232.metafits -n 100 -s srclist_pumav3_EoR0aegean_EoR1pietro+ForA_phase1+2.txt
# WARNING:mwa_pb.beam_full_EE: Calculating for 19812 points. This may take a while!
Traceback (most recent call last):
  File "srclist_by_beam.py", line 164, in <module>
    freqcent=freqcent,delays=delays)
  File "/group/mwa/software/srclists/srclists/srclist_by_beam_lib.py", line 345, in get_beam_weights
    XX,YY = primary_beam.MWA_Tile_full_EE([za], [az], freq=freqcent, delays=delays, zenithnorm=True, power=True, interp=False)
  File "/group/mwa/software/mwa_pb/python3_test/mwa_pb/primary_beam.py", line 54, in MWA_Tile_full_EE
    j = mybeam.get_response(az, za)
  File "/group/mwa/software/mwa_pb/python3_test/mwa_pb/beam_full_EE.py", line 372, in get_response
    Jones = self.get_FF(phi_arr, theta_arr, grid=False)
  File "/group/mwa/software/mwa_pb/python3_test/mwa_pb/beam_full_EE.py", line 541, in get_FF
    emn_T = (1.0j) ** N * (P_sin * (M_u * Q2 - M * Q1) + Q2 * P1) * phi_const
ValueError: operands could not be broadcast together with shapes (19812,1248) (624,)
```
Looks like it's a mwa_pb errror though.

Since I was testing I thought I might as well make a test script which I'll make a pull request for.

**Comment by NickSwainston** [comment]
Can I please have push permissions so I can push my branch

**Comment by cjordan** [comment]
You should fork this repo on github and push your branch there.

**Comment by cjordan** [comment]
I believe this was fixed a while ago (I think it was commit 6d55490).
